eis_toolkit/vector_processing/rasterize_vector.py

```diff
@@ -0,0 +1,125 @@
+from typing import Optional, Tuple, Union
+
+import geopandas as gpd
+import numpy as np
+from rasterio import features, profiles, transform
+
+from eis_toolkit import exceptions
+
+
+def rasterize_vector(
+    geodataframe: gpd.GeoDataFrame,
+    resolution: float,
+    value_column: Optional[str] = None,
+    default_value: float = 1.0,
+    fill_value: float = 0.0,
+    base_raster_profile: Optional[Union[profiles.Profile, dict]] = None,
+    buffer_value: Optional[float] = None,
+) -> Tuple[np.ndarray, dict]:
+    """Transform vector data into raster data.
+
+    Args:
+        geodataframe (geopandas.GeoDataFrame): The vector dataframe to be rasterized.
+        resolution (float): The resolution i.e. cell size of the output raster
+        value_column (Optional[str]): The column name with values for each geometry.
+            If None, then default_value is used for all geometries.
+        default_value (float): Default value burned into raster cells based on geometries.
+        base_raster_profile (Optional[Union[rasterio.profiles.Profile, dict]]): Base raster profile
+            to be used for determining the grid on which vectors are burned in.
+        fill_value (float): VAlue used outside the burned/rasterized geometry cells.
+        buffer_value (Optional[float]): For adding a buffer around passed
+            geometries before rasterization.
+
+    Returns:
+        out_raster (tuple(numpy.ndarray, dict)): Rasterized vector data and metadata.
```

Comment:
You could remove the type annotations and output value name from here. There seems to be also a typo on line 29: "VAlue" instead of "Value".